compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -338,3 +375,59 @@ pub fn create_compressed_metadata_file(
 
     file.write().unwrap()
 }
+
+/// * Xcoff - AIX target does not support arbitrary section names, the metadata
+///   bytes are stored in the `.info` section that has the `STYP_INFO` flag.
+///   A global symbol is required to prevent the linker from throwing away the
+///   metadata section.
+pub fn create_rmeta_file_xcoff(Object: file, data: &[u8]) -> (Vec<u8>, MetadataPosition) {
+    assert!(file.format() == BinaryFormat::Xcoff);
+    file.add_section(
+        file.segment_name(StandardSegment::Text).to_vec(),
+        b".text".to_vec(),
+        SectionKind::Text,
+    );
+    let data_section = file.add_section(
+        file.segment_name(StandardSegment::Data).to_vec(),
+        b".data".to_vec(),
+        SectionKind::Data,
+    );
+    let section = file.add_section(
+        file.segment_name(StandardSegment::Debug).to_vec(),
+        b".info".to_vec(),
+        SectionKind::Debug,
+    );
+    file.add_file_symbol(".file".into());
```

Comment:
The documentation says that `.file` is used when there is a file auxiliary symbol. Ideally all handling of `.file` would be internal to the `object` crate, so here we would specify the actual file name we want.